src/components/featured-map-card/featured-map-card-component.jsx

```diff
@@ -35,31 +36,47 @@ const FeaturesMapCardComponent = ({
 
   const isFeatureMapCardVisible = isOnScreen;
 
-  return (featuredMap && (
-    <div className={cx(className, styles.cardContainer, { [animationStyles.leftHidden]: !isFeatureMapCardVisible, [styles.delayOnOut]: isFeatureMapCardVisible })}>
-      {featuredMap && 
-        <>
-          <section
-            className={styles.titleSection}
-            style={ {backgroundImage: `linear-gradient(rgba(0,0,0,0.3),rgba(0,0,0,0.3)), url(${featuredMap.image})`}}
-          >
-            <ShareModalButton theme={{ shareButton: styles.shareButton}} />
-            <h2 className={styles.title}>{featuredMap.title}</h2>
-          </section>
-          <section className={styles.descriptionSection}>
-            <p className={styles.description}>{featuredMap.description}</p>
-            <button
-              className={styles.allMapsButton}
-              onClick={handleClick}
+  const [isShareModalOpen, setShareModalOpen] = useState(false);
+
+  return (
+    (featuredMap && (
+      <div
+        className={cx(className, styles.cardContainer, {
+          [animationStyles.leftHidden]: !isFeatureMapCardVisible,
+          [styles.delayOnOut]: isFeatureMapCardVisible
+        })}
+      >
+        {featuredMap && (
+          <>
+            <section
+              className={styles.titleSection}
+              style={{
+                backgroundImage: `linear-gradient(rgba(0,0,0,0.3),rgba(0,0,0,0.3)), url(${featuredMap.image})`
+              }}
             >
-              <span className={styles.buttonText}>All maps</span>
-              <ChevronIcon className={styles.arrowIcon}/>
-            </button>
-          </section>
-        </>
-      }
-    </div>) || null
-  )
+              <ShareModalButton
+                theme={{ shareButton: styles.shareButton }}
+                setShareModalOpen={setShareModalOpen}
+              />
+              <ShareModal
+                isOpen={isShareModalOpen}
+                handleClose={() => setShareModalOpen(false)}
```

Comment:
I think a better approach here would be to pass the setter function instead of an arrow function on the render. And then using it on the container of the `ShareModalComponent` passing there the `false` param. That way we would avoid some undesired re render that might happen since this arrow functions [would be created each time](https://reactjs.org/docs/faq-functions.html#arrow-function-in-render).